Remove sideloading of Onyx data for attachment modal

tgolen opened the pull request:
### Details
This PR removes some deprecated methods that were loading data into the components without using Onyx properly.

### Fixed Issues
Part of https://github.com/Expensify/App/issues/27262


### Tests
1. Create a money request using the "scan" method
2. After creating the request you will be on the chat report with a preview of the request
3. Click on the request preview to go to the report details
4. Click on the request again to go to the request details
5. Click on the image of the request
6. Verify that a modal opens and displays the image properly

- [ ] Verify that no errors appear in the JS console

### Offline tests
Same as tests

### QA Steps
Same as tests

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
        - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

![image](https://github.com/Expensify/App/assets/1228807/0f77495a-21f0-4ae1-be3d-b811fbfb2d10)


</details>

<details>
<summary>Android: mWeb Chrome</summary>

Note: My emulator is not able to load the site in chrome due to URLs not loading and HOSTs setup needing to be fixed

</details>

<details>
<summary>iOS: Native</summary>

Note: our local emulators do not show receipts due to our WARP client that we have installed.
![image](https://github.com/Expensify/App/assets/1228807/49d01dfb-21a9-446d-a960-6f4fe42afc7a)

</details>

<details>
<summary>iOS: mWeb Safari</summary>
![image](https://github.com/Expensify/App/assets/1228807/839e889a-441b-49a3-b715-1235ce44c3e5)


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->
![image](https://github.com/Expensify/App/assets/1228807/2a62e3b2-f3b8-4e21-a53f-fc78e25bccb9)

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->
![image](https://github.com/Expensify/App/assets/1228807/1acc255a-d2ca-4003-b128-42ecdb9b5781)

</details>

tgolen committed:
Remove sideloading of Onyx data for attachment modal

situchan commented:
I can help if requires C+ review

tgolen commented:
@situchan That would be helpful, yes please!

tgolen commented:
@situchan would you be able to review this, please? Also bumping @dangrous for review.

situchan commented on src/components/Attachments/AttachmentCarousel/index.js:
```diff
@@ -224,11 +223,29 @@ AttachmentCarousel.propTypes = propTypes;
             canEvict: false,
         },
+        parentReport: {
+            key: ({report}) => `${ONYXKEYS.COLLECTION.REPORT}${report ? report.parentReportID : '0'}`,
+        },
+        parentReportActions: {
+            key: ({report}) => `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${report ? report.parentReportID : '0'}`,
+            canEvict: false,
+        },
+    }),
+
+    // eslint-disable-next-line rulesdir/no-multiple-onyx-in-file
+    withOnyx({
+        transaction: {
+            key: ({report, parentReportActions}) => {
+                const parentReportAction = lodashGet(parentReportActions, [report.parentReportActionID]);
+                return `${ONYXKEYS.COLLECTION.TRANSACTION}${lodashGet(parentReportAction, 'originalMessage.IOUTransactionID', 0)}`;
+            },
```
Regarding these changes, can you confirm that `parentReportID`, `originalMessage.IOUTransactionID` can never be empty string? Because if empty string, it will return all reportActions, transactions

situchan reviewed (commented):
Code looks good to me

situchan commented on src/components/Attachments/AttachmentCarousel/index.native.js:
```diff
@@ -171,11 +170,29 @@ AttachmentCarousel.propTypes = propTypes;
 
 export default compose(
+    // eslint-disable-next-line rulesdir/no-multiple-onyx-in-file
     withOnyx({
         reportActions: {
             key: ({report}) => `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${report.reportID}`,
             canEvict: false,
         },
+        parentReport: {
+            key: ({report}) => `${ONYXKEYS.COLLECTION.REPORT}${report ? report.parentReportID : '0'}`,
+        },
+        parentReportActions: {
+            key: ({report}) => `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${report ? report.parentReportID : '0'}`,
+            canEvict: false,
+        },
+    }),
+
+    // eslint-disable-next-line rulesdir/no-multiple-onyx-in-file
+    withOnyx({
+        transaction: {
+            key: ({report, parentReportActions}) => {
+                const parentReportAction = lodashGet(parentReportActions, [report.parentReportActionID]);
+                return `${ONYXKEYS.COLLECTION.TRANSACTION}${lodashGet(parentReportAction, 'originalMessage.IOUTransactionID', 0)}`;
+            },
+        },
```
I see that https://github.com/Expensify/App/pull/28866 is completely removing multiple `withOnyx`.

dangrous commented on src/components/Attachments/AttachmentCarousel/index.js:
```diff
@@ -224,11 +223,29 @@ AttachmentCarousel.propTypes = propTypes;
+    }),
+
+    // eslint-disable-next-line rulesdir/no-multiple-onyx-in-file
+    withOnyx({
```
Why do these need to be separate? Can't we combine them into one and save the error?

tgolen replied:
This is because it's how dependent Onyx keys work. However, with the latest Onyx update, this is no longer necessary and I'll update this.

dangrous commented on src/components/Attachments/AttachmentCarousel/attachmentCarouselPropTypes.js:
```diff
@@ -20,11 +21,23 @@ const propTypes = {
+    /** The parent of `report` */
+    parentReport: reportPropTypes,
+
+    /** The specific action that links to the parent report */
```
```suggestion
    /** The specific actions that link to the parent report */
```
Unless I'm misunderstanding this. If we're only sending one action, can we send the action itself instead of this container?

tgolen replied:
It's not possible to get the single action via `withOnyx()` so the component must receive all the parent report actions and then lookup the specific report action with `const parentReportAction = parentReportActions[report.parentReportActionID];`

tgolen replied:
I'll update the comment for this prop

dangrous reviewed (changes requested):
Couple of questions, following the questions @situchan asked as well.

tgolen committed:
Merge branch 'main' into tgolen-improve-proploading

tgolen committed:
Remove nested withOnyx

tgolen committed:
Update comment to be accurate

tgolen commented:
OK, I have updated this PR to remove the nested `withOnyx()` calls and to update a comment to be more accurate. This is ready for review again.

dangrous reviewed (dismissed):
So I'm getting a 404 error when following the testing steps but it's requesting a specific png so I think it's an unrelated issue I'll figure out separately. If @situchan can test successfully, that works for me! Code updates look good.

situchan commented on src/components/Attachments/AttachmentCarousel/attachmentCarouselPropTypes.js:
```diff
@@ -20,11 +21,23 @@ const propTypes = {
+    parentReport: reportPropTypes,
+
+    /** The report actions of the parent report */
+    parentReportActions: PropTypes.shape(reportActionPropTypes),
```
This kind of mistake is easy to miss
```suggestion
    parentReportActions: PropTypes.objectOf(PropTypes.shape(reportActionPropTypes)),
```

tgolen replied:
Yes, thank you. I went back and forth on `parentReportAction` and `parentReportActions` and forgot to update this.

situchan commented on src/components/Attachments/AttachmentCarousel/extractAttachmentsFromReport.js:
```diff
@@ -10,11 +10,13 @@ import CONST from '@src/CONST';
  * Constructs the initial component state from report actions
  * @param {Object} report
+ * @param {Object} parentReportAction
  * @param {Array} reportActions
```
I am not sure why this is array

situchan commented on src/components/Attachments/AttachmentCarousel/index.js:
```diff
@@ -226,11 +225,25 @@ AttachmentCarousel.defaultProps = defaultProps;
 AttachmentCarousel.displayName = 'AttachmentCarousel';
 
 export default compose(
+    // eslint-disable-next-line rulesdir/no-multiple-onyx-in-file
```
This comment is no longer needed

tgolen committed:
Fix proptypes, remove unused parameter

tgolen commented:
@situchan I have addressed your comments. I found another proptype that needed updated and I also found an unused param `report` that could be cleaned up.

tgolen commented:
@situchan bump for review please

situchan commented:
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>


https://github.com/Expensify/App/assets/108292595/33460ba7-546a-4d04-a382-f63dd9acacc3



</details>

<details>
<summary>Android: mWeb Chrome</summary>


https://github.com/Expensify/App/assets/108292595/f2cab59c-445c-4972-b1c1-800d7c9e690c



</details>

<details>
<summary>iOS: Native</summary>


https://github.com/Expensify/App/assets/108292595/c98d7cae-828b-48bb-b997-738ae3fc3189



</details>

<details>
<summary>iOS: mWeb Safari</summary>


https://github.com/Expensify/App/assets/108292595/de761438-f6c5-47ec-a6eb-4534eb5478ab



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


https://github.com/Expensify/App/assets/108292595/b5e0301e-28ab-4246-a912-647e4115a12a



</details>

<details>
<summary>MacOS: Desktop</summary>


https://github.com/Expensify/App/assets/108292595/463531ed-4b63-4ae1-8c01-fbf4be08a33e



</details>

situchan commented:
@tgolen regression: attachment modal header title briefly shows "Attachment"


https://github.com/Expensify/App/assets/108292595/b35099c5-d438-428e-bc86-261a4160f5f5

tgolen committed:
Improve accuracy of attachment header title

tgolen commented:
@situchan Thanks for reporting that. I was largely unable to reproduce the issue. On my emulator, I guess the transition happens so fast that I cannot see it. I made an improvement to the code to hopefully improve the header title so that It does not default to either attachment or receipt until it absolutely knows which one it is.

situchan commented:
> @situchan Thanks for reporting that. I was largely unable to reproduce the issue. On my emulator, I guess the transition happens so fast that I cannot see it. I made an improvement to the code to hopefully improve the header title so that It does not default to either attachment or receipt until it absolutely knows which one it is.

It's known that android emulator is super slow. So if you test on that device, you will clearly notice. Also noticeable on android emulator chrome

situchan commented:
One more: download icon changes to three dot


https://github.com/Expensify/App/assets/108292595/1989d250-4190-4dc3-a01f-eba80648afcc

tgolen committed:
Don't show the buttons until the file type is known

tgolen commented:
OK, I've pushed a fix for that too. I'm a bit stuck. My android build keeps failing (of course, right at the last step), so I've been trying to get it built and tested on Android but haven't had much luck. The third time is the charm, right?

tgolen committed:
Merge branch 'main' into tgolen-improve-proploading

situchan commented:
Now found more serious regression:
Not found page briefly shows while scanning request

https://github.com/Expensify/App/assets/108292595/7e88437b-5026-4f13-8e2c-f2825621533a

situchan commented:
@tgolen single withOnyx PR was reverted. I think we should go back to multiple withOnyx in this PR for now.

tgolen committed:
Merge branch 'main' into tgolen-improve-proploading

tgolen commented:
OK, I agree. I'll switch this to use multiple `withOnyx()` for now. This is a bit lower on my list of priorities, so I might not get to it until later this week.

tgolen committed:
Use multiple withOnyx

tgolen committed:
Merge branch 'main' into tgolen-improve-proploading

tgolen committed:
disable eslint for multiple onyx

tgolen commented:
OK, I've updated this to use multiple withOnyx, so it looks like it behaves a little more expectantly now. Can you please take another look at it @situchan ?

situchan commented:
Looks good now

https://github.com/Expensify/App/assets/108292595/4c2d5c22-8ac7-4d4e-bfec-c209e117cdf1

situchan reviewed (approved):
@dangrous all yours

dangrous reviewed (approved):
Cool, I think this makes sense!

dangrous commented:
I'm going to go ahead and merge, not sure why Melvin tagged @lakchote

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6869840079) to staging by https://github.com/dangrous in version: 1.3.99-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6882673474) to production by https://github.com/roryabraham in version: 1.3.99-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6882677947) to staging by https://github.com/dangrous in version: 1.4.0-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅
